pkg/kubectl-argo-rollouts/info/rollout_info.go

```diff
@@ -75,11 +76,21 @@ func NewRolloutInfo(
 	roInfo.Message = message
 	roInfo.Icon = rolloutIcon(roInfo.Status)
 	roInfo.Containers = []*rollout.ContainerInfo{}
-	for c := range ro.Spec.Template.Spec.Containers {
-		curContainer := ro.Spec.Template.Spec.Containers[c]
-		roInfo.Containers = append(roInfo.Containers, &rollout.ContainerInfo{Name: curContainer.Name, Image: curContainer.Image})
+
+	if workloadRef != nil {
+		for c := range workloadRef.Spec.Template.Spec.Containers {
+			curContainer := workloadRef.Spec.Template.Spec.Containers[c]
+			roInfo.Containers = append(roInfo.Containers, &rollout.ContainerInfo{Name: curContainer.Name, Image: curContainer.Image})
+		}
+	} else {
+		for c := range ro.Spec.Template.Spec.Containers {
+			curContainer := ro.Spec.Template.Spec.Containers[c]
+			roInfo.Containers = append(roInfo.Containers, &rollout.ContainerInfo{Name: curContainer.Name, Image: curContainer.Image})
+		}
```

Comment:
I guess we can improve the code by

```
var containerList []corev1.container
if workloadRef != nil {
    containerList = workloadRef.Spec.Template.Spec.Containers
} else {
    containerList = ro.Spec.Template.Spec.Containers
}

for _, c := range ro.Spec.Template.Spec.Containers {
	roInfo.Containers = append(roInfo.Containers, &rollout.ContainerInfo{Name: c.Name, Image: c.Image})
}
```

Reply:
Ah, I did try this, but then there is a panic at `for _, c := range ro.Spec.Template.Spec.Containers {` if using a workloadRef, because the rollout has no containers.

I also considered setting a "workload" variable to either the rollout or deployment depending on if using workloadRef, but the issue there is they are not the same type.

Reply:
My mistake. The for loop should be `for _, c := range containerList {`. WDYT?

Reply:
Ah I see good suggestion, changing now! Thanks